packages/flutter/lib/src/cupertino/text_selection.dart

```diff
@@ -446,5 +452,675 @@ class _CupertinoTextSelectionControls extends TextSelectionControls {
+
+    // Update list children.
+    _children = updateChildren(_children, widget.children, forgottenChildren: _forgottenChildren);
+    _forgottenChildren.clear();
+  }
+}
+
+// The custom RenderBox that helps paginate the menu items.
+class _CupertinoTextSelectionToolbarItemsRenderBox extends RenderBox with ContainerRenderObjectMixin<RenderBox, ToolbarItemsParentData>, RenderBoxContainerDefaultsMixin<RenderBox, ToolbarItemsParentData> {
+  _CupertinoTextSelectionToolbarItemsRenderBox({
+    @required double dividerWidth,
+    @required int page,
+  }) : assert(dividerWidth != null),
+       assert(page != null),
+       _dividerWidth = dividerWidth,
+       _page = page,
+       super();
+
+  final Map<_CupertinoTextSelectionToolbarItemsSlot, RenderBox> slotToChild = <_CupertinoTextSelectionToolbarItemsSlot, RenderBox>{};
+  final Map<RenderBox, _CupertinoTextSelectionToolbarItemsSlot> childToSlot = <RenderBox, _CupertinoTextSelectionToolbarItemsSlot>{};
+
+  RenderBox _updateChild(RenderBox oldChild, RenderBox newChild, _CupertinoTextSelectionToolbarItemsSlot slot) {
+    if (oldChild != null) {
+      dropChild(oldChild);
+      childToSlot.remove(oldChild);
+      slotToChild.remove(slot);
+    }
+    if (newChild != null) {
+      childToSlot[newChild] = slot;
+      slotToChild[slot] = newChild;
+      adoptChild(newChild);
+    }
+    return newChild;
+  }
+
+  int _page;
+  int get page => _page;
+  set page(int value) {
+    if (value == _page) {
+      return;
+    }
+    _page = value;
+    markNeedsLayout();
+  }
+
+  RenderBox _backButton;
+  RenderBox get backButton => _backButton;
+  set backButton(RenderBox value) {
+    _backButton = _updateChild(_backButton, value, _CupertinoTextSelectionToolbarItemsSlot.backButton);
+  }
+
+  double _dividerWidth;
+  double get dividerWidth => _dividerWidth;
+  set dividerWidth(double value) {
+    if (value == _dividerWidth) {
+      return;
+    }
+    _dividerWidth = value;
+    markNeedsLayout();
+  }
+
+  RenderBox _nextButton;
+  RenderBox get nextButton => _nextButton;
+  set nextButton(RenderBox value) {
+    _nextButton = _updateChild(_nextButton, value, _CupertinoTextSelectionToolbarItemsSlot.nextButton);
+  }
+
+  RenderBox _nextButtonDisabled;
+  RenderBox get nextButtonDisabled => _nextButtonDisabled;
+  set nextButtonDisabled(RenderBox value) {
+    _nextButtonDisabled = _updateChild(_nextButtonDisabled, value, _CupertinoTextSelectionToolbarItemsSlot.nextButtonDisabled);
+  }
+
+  @override
+  void performLayout() {
+    if (firstChild == null) {
+      performResize();
+      return;
+    }
+
+    // Layout slotted children.
+    _backButton.layout(constraints.loosen(), parentUsesSize: true);
+    _nextButton.layout(constraints.loosen(), parentUsesSize: true);
+    _nextButtonDisabled.layout(constraints.loosen(), parentUsesSize: true);
+
+    double buttonPosition = 0.0;
+    double parentWidth = constraints.maxWidth; // The width of the whole widget.
```

Comment:
maybe: toolbarWidth?

Reply:
Why does it start out as the max width possible instead of zero?

Reply:
I think I was using it as a constraint earlier and then I changed my code, good call.